### ISSUE TITLE
Latest version working with arduCam Hawkeye 64mp?

### ISSUE THREAD

**Issue by sam-mcelhinney** [Gordon999/Pi_LIbCamera_GUI#14]
hi Gordon

I’ve been using your pi_libCamera_gui for a while with a home made telescope that I’ve been building, using a Pi HQ camera. It works great!

But, I just got myself an Arducam 64mp camera, and wanted to try it with that. Oddly, whilst v1.1 sort of seems to work (it recognises it as a picam v3, and that causes issues with the image tho) the newer v4.52 doesn’t display any preview image at all.

Am I missing something I need to do to set it up properly?
Sam

Edit - I have all the necessary drivers etc now installed and the usual libCamera-hello etc works well with the camera

**Comment by Gordon999** [comment]
I don't have a arducam 64mp to test.
What dimensions (w and h) does the file test.jpg show? To say its a v3 it's presumably 4608 pixels wide, when it expects 9152 or 4624, which used to work.

**Comment by sam-mcelhinney** [comment]
Hi there! Thanks for the reply 

I’ve tested on both versions again… 

- Both have a test.jpg image width of 9152, so correct size. Good news there I guess.
- V1.1 the image in the jpg is very blurred, and zoomed in it seems, so no good.
- V4.52 the jpg image looks good, camera focuses well, lighting is balanced.

but… when the GUI opens with v4.52, it says it recognises the camera as 64mp; but never displays the preview. The top left message always remains at seeking preview.

happy to run any tests at all - I have no python knowledge (just c++ so far) but can give it a go/learn.

thanks again
Sam

**Comment by Gordon999** [comment]
Are you using the standard pi libcamera or the arducam version ?

I will upload a beta version that mostly works with the pi version to try. Probably tomorrow.

**Comment by sam-mcelhinney** [comment]
Pretty sure I have all the arducam libcamera installed, I worked through their user instructions, and their camera works well with libcamera calls. (It did take a while to get right tho). I could reinstall them again, see if it makes a difference.

No rush, I’ve been working on the scope for two years now, so happy to be patient!

**Comment by Gordon999** [comment]
I suspect you are using the default Pi version of Libcamera. I find using the arducam version a pain as you can't get Pi updates.
Try the BETA version of PiLibcameraGUI_Ard.py l have uploaded. It should work with the default libcamera BUT has no AUTO Focus only manual.
Ensure you have the dtoverlay in /boot/config.txt.  dtoverlay=arducam-64mp ?

I have included a rather crude AF ( more work needed). Use AUTO Focus mode and Zoom in. Click on image to move AF point.

I have also added the option of adding 2 manual focus buttons , gpios 16 and 21. connect to gnd
When you use them, or the screen manual focus button, focus for a maximum value on the onscreen focus value

**Comment by sam-mcelhinney** [comment]
It works, super!

There is a slight issue around the right and bottom edge of the buttons which looks like a setting out calc, but if I size my preview width much higher (to try and close the gap) it causes the buttons to fail altogether. But no big deal. Maybe one day I’ll get a little more into the python code to tinker with that anyway

Thank you for your help!

S

![05969916-F3DF-4505-8D4E-DE04ABB50EC5](https://github.com/Gordon999/Pi_LIbCamera_GUI/assets/27627482/04fa7466-4a6b-41ce-aa3a-bebdd506cc38)

**Comment by Gordon999** [comment]
Great that it works. 
I will  to work on a AF algorithm to avoid using the Arducam Libcamera version.

**Comment by sam-mcelhinney** [comment]
Hi Gordon
All good, I fixed the button issue by incrementally increasing the width of the preview. Something odd in how I worked out my screen resolution, I think.

I don’t actually need the autofocus function, as I’m building the camera into a Newtonian ‘scope, so spent the last couple of nights carefully removing the lens down to the sensor. It is still working, although the replacement IR cut I’ve put in isn’t filtering as much as I thought… always more to work on.

Thanks for all your help - one day I might get into the python code itself and make a few tweaks to reflect what I am using it for a little better

Sam

![31E2C0AC-3525-4EED-9183-2204C01B88F7](https://github.com/Gordon999/Pi_LIbCamera_GUI/assets/27627482/0f89641b-0a0b-4c61-b044-790208b4c7d7)


![EC1ADD46-DAB0-496A-A93A-FB8BB9035B71](https://github.com/Gordon999/Pi_LIbCamera_GUI/assets/27627482/19fd89fb-265d-430e-b701-341f937f6c44)
